reader_concurrency_semaphore.cc

```diff
@@ -976,7 +976,10 @@ future<> reader_concurrency_semaphore::execution_loop() noexcept {
             tracing::trace(permit.trace_state(), "[reader concurrency semaphore {}] executing read", _name);
 
             try {
+                utils::latency_counter lc;
+                lc.start();
                 e.func(reader_permit(permit.shared_from_this())).forward_to(std::move(e.pr));
+                _reads().mark(lc.stop().latency());
```

Comment:
It is not guaranteed that the read has finished at this point. It can be simply preempted. In most reads with bad latencies will have just gotten started at this point. If you want to stop the latency counter *after* the read has finished, you have to add a continuation (although Avi will protest loudly if you do).

Reply:
Also, you are covering only part of the reads here, although it is the important part -- the one we really care about.

Reply:
Yes, I realized that this morning that it will not work the way I initially thought. Just for testing purposes, I switched to the continuation.